test/plots/xyPlotTests.ts

```diff
@@ -2,268 +2,348 @@
         plot.destroy();
 
-        assert.strictEqual((<any> xScale)._callbacks.size, 0, "the plot is no longer attached to xScale");
-        assert.strictEqual((<any> yScale)._callbacks.size, 0, "the plot is no longer attached to yScale");
+        xScale.domain([0, 2]);
+
+        assert.deepEqual(yScale.domain(), [0, 1], "autoranging no longer occurs");
 
         svg.remove();
       });
     });
 
-    describe("Automatic domain and range adjustment for X and Y Scales", () => {
-      let SVG_HEIGHT = 400;
-      let SVG_WIDTH = 400;
-      let svg: d3.Selection<void>;
+    describe("deferred rendering", () => {
       let xScale: Plottable.Scales.Linear;
       let yScale: Plottable.Scales.Linear;
       let plot: Plottable.XYPlot<number, number>;
-      let dataset: Plottable.Dataset;
-      let defaultInterval = [0, 1];
 
       beforeEach(() => {
-        svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
         xScale = new Plottable.Scales.Linear();
         yScale = new Plottable.Scales.Linear();
-        dataset = new Plottable.Dataset([
-          { a: -6, b: 6 },
-          { a: -2, b: 2 },
-          { a: 2, b: -2 },
-          { a: 6, b: -6 }
-        ]);
         plot = new Plottable.XYPlot<number, number>();
-        (<any> plot)._createDrawer = (dataset: Plottable.Dataset) => createMockDrawer(dataset);
-        plot.addDataset(dataset);
       });
 
-      it("automatically adjusts domain and range after rendering to DOM", () => {
-        plot.x((d: any) => d.a, xScale)
-            .y((d: any) => d.b, yScale);
-        assert.deepEqual(xScale.domain(), defaultInterval, "X domain is not changed before rendering");
-        assert.deepEqual(xScale.range(), defaultInterval, "X range is not changed before rendering");
-        assert.deepEqual(yScale.domain(), defaultInterval, "Y domain is not changed before rendering");
-        assert.deepEqual(yScale.range(), defaultInterval, "Y range is not changed before rendering");
-
-        plot.renderTo(svg);
-        assert.deepEqual(xScale.domain(), [-7, 7], "X domain has been set automatically after rendering");
-        assert.deepEqual(xScale.range(), [0, SVG_WIDTH], "X range has been set automatically after rendering");
-        assert.deepEqual(yScale.domain(), [-7, 7], "Y domain has been set automatically after rendering");
-        assert.deepEqual(yScale.range(), [SVG_HEIGHT, 0], "Y range has been set automatically after rendering");
+      it("can set if rendering is deferred", () => {
+        assert.strictEqual(plot.deferredRendering(), false, "deferred rendering is false by default");
+        assert.strictEqual(plot.deferredRendering(true), plot, "setting the deferred rendering option returns the plot");
+        assert.strictEqual(plot.deferredRendering(), true, "deferred rendering can be turned on");
+        plot.deferredRendering(false);
+        assert.strictEqual(plot.deferredRendering(), false, "deferred rendering can be turned off");
+      });
+    });
```

Comment:
deferredRendering logic essentially is 2 parts:
* Temporarily have the elements in the right locations but not fully correct attributes
* After a timeout, fully render.

I can attempt to test the first piece, but since having timeouts in tests is a bad practice, I do not think I can do the second.  The awkward part about this piece is that I will essentially be testing for intermediate "incorrect" behavior.  Is this ok?